simulator-ui/tests/entities.spec.ts

```diff
@@ -0,0 +1,206 @@
+      { testId: 'testParameterEntityTestResultLink', expectedValue: '0' },
+      { testId: 'testParameterEntityCreatedDate', expectedValue: exampleDate },
+      { testId: 'testParameterEntityLastModifiedDate', expectedValue: exampleDate },
+    ],
+    testIdToBeVisible: ['', ''],
+  },
+];
+
+entityPageContentMap.forEach((contentObject: EntityPageContentObject) => {
+  test(`${contentObject.testName}`, async ({ page }) => {
+    // 'first test'
+    await mockBackendResponse(page, contentObject.apiUrl, contentObject.contentJson);
+
+    await page.goto(contentObject.entityUrl);
+
+    await expect(page.locator('th :text("ID")').nth(0)).toHaveCount(1);
+    await checkEntityPageContentValueAndVisibility(page, contentObject);
+    // 'second test'
+    await checkIfRefreshButtonWorks(page, contentObject);
+  });
+});
+
+test('should show message headers when clicking button on message row', async ({ page }) => {
+  const contentObject = entityPageContentMap[0];
+  await mockBackendResponse(page, contentObject.apiUrl, contentObject.contentJson);
+
+  await page.goto(contentObject.entityUrl);
+
+  await page.getByTestId('filterOtherEntityButton').nth(0).click();
+  await expect(page.getByTestId('filterValue')).toHaveText('messageId.in: 1');
+  await expect(page).toHaveURL('http://localhost:9000/message-header?filter%5BmessageId.in%5D=1');
+});
+
+test('should show test parameters when clicking on button in test results row', async ({ page }) => {
+  const contentObject = entityPageContentMap[4];
+  await mockBackendResponse(page, contentObject.apiUrl, contentObject.contentJson);
+
+  await page.goto(contentObject.entityUrl);
+
+  await page.getByTestId('testParametersButton').click();
+  await expect(page.getByTestId('filterValue')).toHaveText('testResultId.in: 1');
+  await expect(page).toHaveURL('http://localhost:9000/test-parameter?filter%5BtestResultId.in%5D=1');
+});
+
+test('should display table of test parameters', async ({ page }) => {
+  const contentObject = entityPageContentMap[5];
+  await mockBackendResponse(page, contentObject.apiUrl, contentObject.contentJson);
+
+  await page.goto(contentObject.entityUrl);
+
+  await checkEntityPageContentValueAndVisibility(page, contentObject);
+  await checkIfRefreshButtonWorks(page, contentObject);
+});
+
+const checkEntityPageContentValueAndVisibility = async (page: Page, contentObject: EntityPageContentObject): Promise<void> => {
+  for (const locator of contentObject.locators) {
+    await expect(page.locator(locator)).toHaveCount(1);
+  }
+  for (const testId of contentObject.testIdsAndExpectedValues) {
+    await expect(page.getByTestId(testId.testId)).toHaveText(testId.expectedValue);
+  }
+  for (const testId of contentObject.testIdToBeVisible) {
+    await expect(page.getByTestId(testId)).toBeVisible();
+  }
+};
+
+const checkIfRefreshButtonWorks = async (page: Page, contentObject: EntityPageContentObject): Promise<void> => {
```

Comment:
I personally prefer "verify" instead of "check", but that's a matter of taste. I don't expect you to change it.